File: mongodb/documentation/source_code/task6/1g.java
/*
private void executeTask1g() {
    AggregateIterable<Document> aggregateIterable = businessCollection.aggregate(
            asList(
                    lookup("Review", "business_id", "business_id", "Reviews"),
                    unwind("$Reviews"),
                    group("$name", avg("averageStars", "$stars")),
                    match(lt("averageStars", 3))
            )
    );

    for (Document document : aggregateIterable) {
        Document documentToBeDeleted = new Document(new Document("name", document.getString("_id")));
        DeleteResult deleteResult = businessCollection.deleteMany(documentToBeDeleted);
        System.out.println(deleteResult);
    }
}
*/

File: mongodb/src/main/java/mongo_solutions/1g.js
var businesses = db.Business.aggregate([
    {$lookup: {from: "Review", localField: "business_id", foreignField: "business_id", as: "Reviews"}},
    {$unwind: "$Reviews"},
    {$group: {_id: "$name", averageStars: {$avg: "$stars"}}},
    {$sort: {averageStars: -1}}
]);

businesses.forEach(function (business) {
    if (business.avg < 3) {
        db.Business.remove({"name": business._id});
    }
});